functions/authenticate.js

```diff
@@ -0,0 +1,123 @@
+// Use this code snippet in your app.
+// If you need more information about configurations or implementing the sample code, visit the AWS docs:
+// https://aws.amazon.com/developers/getting-started/nodejs/
+
+// Load the AWS SDK
+const url = require('url');
+const https = require('https')
+const { getSecret, updateSecret } = require('./secretManager.js');
+
+function getReason(err) {
+  if (err)
+    return err.message;
+  else
+    return '';
+}
+
+async function sendResponse(event, context, status, err) {
+	return new Promise((resolve, reject) => {
+		// only send response when called via CFT
+		if (!event.ResponseURL || !event.StackId || !event.RequestId || !event.LogicalResourceId) {
+			return;
+		}
+		
+		var responseBody = {
+			StackId: event.StackId,
+			RequestId: event.RequestId,
+			LogicalResourceId: event.LogicalResourceId,
+			PhysicalResourceId: context.logStreamName,
+			Status: status,
+			Reason: getReason(err) + " See details in CloudWatch Log: " + context.logStreamName,
+		};
+
+		console.log("RESPONSE:\n", responseBody);
+		var json = JSON.stringify(responseBody);
+		var parsedUrl = url.parse(event.ResponseURL);
+		var options = {
+			hostname: parsedUrl.hostname,
+			port: 443,
+			path: parsedUrl.path,
+			method: "PUT",
+			headers: {
+				"content-type": "",
+				"content-length": json.length
+			}
+		};
+
+		var request = https.request(options, function(response) {
+			console.log("STATUS: " + response.statusCode);
+			console.log("HEADERS: " + JSON.stringify(response.headers));
+			context.done(null, null);
+		});
+
+		request.on("error", function(error) {
+			console.log("sendResponse Error:\n", error);
+			context.done(error);
+		});
+
+		request.on("end", function() {
+			console.log("end");
+			resolve();
+		});
+		
+		request.write(json);
+		request.end();
+	});
+}
+
+async function getNewAccessToken(authUrl, client_id, user_key) {
+	let apiAccessAuthOptions = url.parse(authUrl);
```

Comment:
Let's ensure this works with alpha as well as staging cloud URLs.

Reply:
I get a `Invalid argument 'accessTokenRequest.grant_type'` error when I call the token api's in Alpha. Will have to check this.